torchgeo/datasets/geo.py

```diff
@@ -640,6 +654,10 @@ def __init__(
             else:
                 mint = 0
                 maxt = sys.maxsize
+                match = re.match(filename_regex, os.path.basename(filepath))
```

Comment:
This needs a:
```python
if match:
```
right?

Reply:
Let's move the re.match outside of the try-except, similar to RasterDataset. Then we can skip files that don't match the regex.

Reply:
yep, fixed